asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py

```diff
@@ -60,6 +61,115 @@ def to_solvent_component(self) -> gufe.SolventComponent:
+    )
+    solvent_padding_complex: FloatQuantity["nanometer"] = Field(
+        1.0 * OFFUnit.nanometer,
+        description="The solvent padding (in nm) to use for the complex phase of each edge.",
+    )
+    solvent_padding_solvated: FloatQuantity["nanometer"] = Field(
+        1.2 * OFFUnit.nanometer,
+        description="The solvent padding (in nm) to use for the solvated phase of each edge.",
+    )
+
+    def get_adapted_sampling_protocol(
+        self, scorer_method, mapping, protocol, base_sampling_length
+    ):
+        """
+        It's advisable to increase simulation time on edges that are expected to be less reliable. There
+        Aren't many good estimators for this, but the network planner edge scoring is a decent approximation.
+
+        If the edge scoring (computed using `scorer_method`) is below the `adaptive_sampling_threshold` the
+        simulation time is multiplied by `adaptive_sampling_multiplier`. Just to be sure, we use the base
+        protocol's sampling time and not the provided edge protocol sampling time as a base value.
+
+        Returns the adjusted OpenFE Protocol.
+        """
+        if scorer_method == "default_lomap":
+            scorer = lomap_scorers.default_lomap_score
+        elif scorer_method == "default_perses":
+            scorer = perses_scorers.default_perses_scorer
+        else:
+            raise ValueError(
+                f"Atom mapping scorer {scorer_method} not recognized; use one of `default_lomap`, `default_perses`."
+            )
+        if scorer(mapping) < self.adaptive_sampling_threshold:
+            protocol._settings.simulation_settings.production_length = (
+                base_sampling_length * self.adaptive_sampling_multiplier
+            )
+        return protocol
+
+    def get_adapted_solvent_protocol(self, leg, protocol):
+        """
+        Certain water box shapes (such as dodecahedron) are able to handle slightly smaller padding size
+        in the complex phase compared to the solvated phase. Given the leg (either "solvent" or "complex")
+        this method applies the specified padding per phase (`solvent_padding_solvated` or
+        `solvent_padding_complex`, resp.).
+
+        Returns the adjusted OpenFE Protocol.
+        """
+        if leg == "solvent":
+            protocol._settings.solvation_settings.solvent_padding = (
+                self.solvent_padding_solvated
+            )
+        else:
+            protocol._settings.solvation_settings.solvent_padding = (
+                self.solvent_padding_complex
+            )
+        return protocol
+
+    def apply_settings(
+        self,
+        edge_protocol: openfe.protocols.openmm_rfe.RelativeHybridTopologyProtocol,
+        network_scorer: str,
+        mapping: "LigandAtomMapping",
+        leg: str,
+        base_protocol: openfe.protocols.openmm_rfe.RelativeHybridTopologyProtocol,
+    ):
```

Comment:
The list of things to pass in is probably going to grow in time should we just pass in the `FreeEnergyCalculationNetwork` and then we can extract everything we need?

Reply:
agreed, although for now I just added a comment so that we can migrate once we're building at this spot again. It's a bit complicated to derive all those parameters just from the `FreeEnergyCalculationNetwork` alone.